services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/ResumableFileDownload.java

```diff
@@ -66,31 +64,25 @@ public boolean equals(Object o) {
         if (!downloadFileRequest.equals(that.downloadFileRequest)) {
             return false;
         }
-        if (!Objects.equals(lastModified, that.lastModified)) {
+        if (!Objects.equals(s3ObjectLastModified, that.s3ObjectLastModified)) {
+            return false;
+        }
+        if (!Objects.equals(fileLastModified, that.fileLastModified)) {
             return false;
         }
-        return Objects.equals(transferSizeInBytes, that.transferSizeInBytes);
+        return Objects.equals(totalSizeInBytes, that.totalSizeInBytes);
     }
 
     @Override
     public int hashCode() {
         int result = downloadFileRequest.hashCode();
         result = 31 * result + (int) (bytesTransferred ^ (bytesTransferred >>> 32));
-        result = 31 * result + (lastModified != null ? lastModified.hashCode() : 0);
-        result = 31 * result + (transferSizeInBytes != null ? transferSizeInBytes.hashCode() : 0);
+        result = 31 * result + (s3ObjectLastModified != null ? s3ObjectLastModified.hashCode() : 0);
+        result = 31 * result + (fileLastModified != null ? fileLastModified.hashCode() : 0);
+        result = 31 * result + (totalSizeInBytes != null ? totalSizeInBytes.hashCode() : 0);
         return result;
     }
 
-    @Override
-    public String toString() {
-        return ToString.builder("ResumableFileDownload")
-                       .add("downloadFileRequest", downloadFileRequest)
-                       .add("bytesTransferred", bytesTransferred)
-                       .add("lastModified", lastModified)
-                       .add("transferSizeInBytes", transferSizeInBytes)
-                       .build();
```

Comment:
This method will be implemented in a separate PR as part of serialization/deserialization implementation